test/road_test.py

```diff
@@ -16,15 +16,27 @@ def test_transition_probs_without_obstacles_are_always_1():
     for a in ACTIONS:
         for next_state, prob, reward in road_test.successors(a):
             assert prob == 1.0
+# -----------------------------------------------------------------------------------------------------
 
-def test_transition_probs_with_one_obstacle_are_1(): #here we test what happens when the car hits/doesn't hit the obstacle
+# The following tests whether we obtain a probability == 1.0 if either a bump or a pedestrian collides or does not collide with the car.
+import pytest
+@pytest.mark.parametrize("obst", [Bump(2, 2)], Pedestrian(2, 2)])
+@pytest.mark.parametrize("car_inst", [Car(3, 2, 1)], Car(3, 1, 1)]) # first will have collision, second will have no collision
+
```

Comment:
Python convention for decorators is to have no whitespace between them and their associated function.